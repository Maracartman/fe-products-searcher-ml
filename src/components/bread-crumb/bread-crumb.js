/* eslint react/no-unescaped-entities: 0 */
import React from 'react';
import './bread-crumb.scss';
import PropTypes from 'prop-types';

export const BreadCrumb = ({ categories }) => (
  <React.Fragment>
    {categories && (
      <div className={'bread-crumb__container'}>
        <p>
          {`${categories.slice(0, categories.length - 1).join(' > ')}`} >{' '}
          <b>{categories[categories.length - 1]}</b>
        </p>
      </div>
    )}
  </React.Fragment>
);

BreadCrumb.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string)
};
