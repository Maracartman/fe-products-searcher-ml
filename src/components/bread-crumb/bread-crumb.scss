.bread-crumb__container {
    margin-top: 1.6rem;

    p {
        color: #a0a0a0;
    }

    b {
        color: #9d9d9d;
    }
}