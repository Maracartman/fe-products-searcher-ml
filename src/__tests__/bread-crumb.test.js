import React from 'react';
import { BreadCrumb } from '../components/bread-crumb/bread-crumb';
import { shallow } from 'enzyme';

describe('BreadCrumb test suite', () => {
  const categories = ['Test1', 'Test2', 'Test3'];
  it('Should not render anything without categories passes', () => {
    const component = shallow(<BreadCrumb categories={categories} />);
    expect(component).toMatchSnapshot();
  });

  it('Should BreadCrumb render expected text', () => {
    const component = shallow(<BreadCrumb categories={categories} />);
    expect(
      component.contains(
        `${categories.slice(0, categories.length - 1).join(' > ')}`
      )
    ).toEqual(true);
  });

   it('Expect BreadCrumb to not render expected text', () => {
    const component = shallow(<BreadCrumb />);
    expect(
      component.contains(
        `${categories.slice(0, categories.length - 1).join(' > ')}`
      )
    ).toEqual(false);
  });
});
