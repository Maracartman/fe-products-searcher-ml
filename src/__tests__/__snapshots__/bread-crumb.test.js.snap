// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb test suite Should not render anything without categories passes 1`] = `ShallowWrapper {}`;
